lampod/src/ln/peer_manager.rs

```diff
@@ -151,18 +152,36 @@ impl LampoPeerManager {
             async_run!(async move {
                 let bind_addr = format!("0.0.0.0:{}", listen_port);
                 log::info!(target: "lampo", "Litening for in-bound connection on {bind_addr}");
-                let listener = tokio::net::TcpListener::bind(bind_addr).await.unwrap();
+                let listener = match tokio::net::TcpListener::bind(bind_addr).await {
+                    Ok(listener) => listener,
+                    Err(e) => {
+                        log::error!(target: "lampo", "Error binding to address: {}", e);
+                        return Err::<(), error::Error>(e.into());
+                    }
```

Comment:
```suggestion
                    Err(e) => error::bail!("Error binding to address: {err}")
```

Reply:
On implementing `error::bail!("Error binding to address: {err}")`the compiler complains about 

```
error[E0282]: type annotations needed
   --> lampod/src/ln/peer_manager.rs:161:17
    |
161 | /                 loop {
162 | |                     let peer_manager = peer_manager.clone();
163 | |
164 | |                     match listener
...   |
182 | |                     }
183 | |                 }
    | |_________________^ cannot infer type

For more information about this error, try `rustc --explain E0282`.
warning: `lampod` (lib) generated 1 warning
```

Could you please tell why is this error is coming, and how can i infer the type of `peer_manager`